Automattic/jetpack
# Google Fonts: update the method used to preconnect Fonts source domain.

Follow-up from #23045
Fixes #23306

#### Changes proposed in this Pull Request:

Instead of manually outputting a link to the `head`, let's rely on a Core filter, `wp_resource_hints`, to take care of that for us.
Let's take that opportunity to ensure we only do that when the Webfonts API is supported on the site.

#### Jetpack product discussion

* N/A

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* See #23045 for the basics.
* Enable the module and check that Google fonts are available.
* Enable and disable the Gutenberg plugin and make sure no errors are logged.
